pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -854,6 +855,29 @@ protected BookieAffinityGroupData internalGetBookieAffinityGroup() {
         }
     }
 
+    public void setNamespaceBundleAffinity (String bundleRange, String destinationBroker) {
+        if (destinationBroker != null) {
+            if (!this.isLeaderBroker()) {
+                LeaderBroker leaderBroker = pulsar().getLeaderElectionService().getCurrentLeader().get();
+                String leaderBrokerUrl = leaderBroker.getServiceUrl();
```

Comment:
this will not work in case of https. we need to select URL based on the request type HTTP or HTTPS? 
you can find reference at: https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java#L104

Reply:
This will work based on incoming requests ( http or https).